lib/spack/spack/package.py

```diff
@@ -1188,6 +1196,25 @@ def do_patch(self):
         else:
             touch(no_patches_file)
 
+    @classmethod
+    def all_patches(cls):
+        """Retrieve all patches associated with the package, including patches
```

Comment:
docstring should start with a short, one-line sentence with a period, and have details below that.